assembly/src/tokens/stream.rs

```diff
@@ -91,6 +96,15 @@ impl<'a> TokenStream<'a> {
         self.pos
     }
 
+    /// Returns the line number of the current token, capped by the last line of the source.
+    pub fn line_number(&self) -> usize {
+        self.lines
+            .get(self.pos)
+            .or_else(|| self.lines.last())
+            .copied()
+            .unwrap_or_default()
```

Comment:
Why do we need to do this capping? Or I guess, under which circumstances can `self.pos` be out of bounds for `self.lines`? Seems like this could be just `self.lines[pos]`.

Reply:
If `eof`, then we are out of bounds. We can simplify as follows:

```
self.lines[self.pos.min(self.lines.len() - 1)]
```